### PR TITLE
Make explicit downcasts of String lengths.

### PR DESCRIPTION
**Pull request by kkoopa**
msvc correctly warns about this

### CODE REVIEW

File: nan_implementation_pre_12_inl.h [hunk 1]
```diff
@@ -185,7 +186,8 @@ Factory<v8::String>::New(const uint8_t * value, int length) {
   if (wideString.size() == 0) {
     return v8::String::Empty();
   } else {
-    return v8::String::New(&wideString.front(), wideString.size());
+    return v8::String::New(&wideString.front()
+         , static_cast<int>(wideString.size()));
```
Comment (bnoordhuis) [inline nan_implementation_pre_12_inl.h]: No assert here?

Comment (kkoopa) [inline nan_implementation_pre_12_inl.h]: I think there's an assert in the widenString function called above, so it should not be necessaey to assert again, as the length ought to have been checked already.

Comment (kkoopa) [inline nan_implementation_pre_12_inl.h]: But I see now that it is only when no length is specified to `widenString`, so yes, should add an assert here.

Comment (kkoopa) [inline nan_implementation_pre_12_inl.h]: Moved the assert in widenString outside the branch instead. Now `wideString.size()` is guaranteed to be `<= INT_MAX` at this point.

File: nan_implementation_pre_12_inl.h [hunk 2]
```diff
@@ -166,9 +167,9 @@ void
 widenString(std::vector<uint16_t> *ws, const uint8_t *s, int l = -1) {
   if (l < 0) {
     size_t foundLength = strlen(reinterpret_cast<const char*>(s));
-    assert(foundLength <= INT_MAX && "string to long");
     l = foundLength;
```
Comment (bnoordhuis) [inline nan_implementation_pre_12_inl.h]: There's an implicit downcast here now.

Comment (kkoopa) [inline nan_implementation_pre_12_inl.h]: crap, you're right

**Comment by kkoopa** [comment]
Hopefully I got it right this time.

File: nan.h [hunk 3]
```diff
@@ -29,6 +29,9 @@
 #include <climits>
 #if defined(_MSC_VER)
 # pragma warning( disable : 4530 )
+# if _MSC_VER < 1700  // < VS 2012, don't know when the warning stops appearing
+#  pragma warning( disable : 4344 )
```
Comment (bnoordhuis) [inline nan.h]: I agree that C4344 is an annoying warning but I don't think disabling it file-wide is a good thing.

File: nan_implementation_pre_12_inl.h [hunk 4]
```diff
@@ -158,18 +158,19 @@ Factory<v8::String>::New(const char * value, int length) {
...
   if (l < 0) {
-    size_t foundLength = strlen(reinterpret_cast<const char*>(s));
-    assert(foundLength <= INT_MAX && "string to long");
-    l = foundLength;
+    len = strlen(reinterpret_cast<const char*>(s));
   }
-  ws->resize(l);
+  assert(len <= INT_MAX && "string too long");
+  ws->resize(len);
   std::copy(s, s + l, ws->begin());
```
Comment (bnoordhuis) [inline nan_implementation_pre_12_inl.h]: s + len?

File: nan.h (hunk 3 above)
Comment (kkoopa) [inline nan.h]: Perhaps not. As it cannot be limited, I'll just remove it.

File: nan_implementation_pre_12_inl.h (hunk 4 above)
Comment (kkoopa) [inline nan_implementation_pre_12_inl.h]: facepalm

**Comment by kkoopa** [comment]
Fourth? time's the charm.

**Pull request merged**